src/Symfony/Component/Messenger/Command/SetupTransportsCommand.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Command;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Messenger\Transport\SetupableTransportInterface;
+
+/**
+ * @author Vincent Touzet <vincent.touzet@gmail.com>
+ */
+class SetupTransportsCommand extends Command
+{
+    protected static $defaultName = 'messenger:setup-transports';
+
+    private $receiverLocator;
+    private $receiverNames;
+
+    public function __construct(ContainerInterface $receiverLocator, array $receiverNames = [])
+    {
+        $this->receiverLocator = $receiverLocator;
+        $this->receiverNames = $receiverNames;
+
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->addArgument('transport', InputArgument::OPTIONAL, 'Name of the transport to setup', null)
+            ->setHelp(<<<EOF
+The <info>%command.name%</info> command setups the transports:
+
+    <info>php %command.full_name%</info>
+
+Or a specific transport only:
+
+    <info>php %command.full_name% <transport></info>
+EOF
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $receiverNames = $this->receiverNames;
+        // do we want to setup only one receiver?
+        if ($transport = $input->getArgument('transport')) {
+            $receiverNames = [$transport];
+        }
+
+        foreach ($receiverNames as $id => $receiverName) {
+            if (!$this->receiverLocator->has($receiverName)) {
+                $io->error(sprintf('The "%s" transport does not exist.', $receiverName));
```

Comment:
This check should be done only when the transport name come from the user input, not in this loop. (and it should fail, not "ignore")